zyte_spider_templates/spiders/ecommerce.py

```diff
@@ -241,8 +257,40 @@ def get_start_request(self, url):
         )
 
     def start_requests(self) -> Iterable[Request]:
-        for url in self.start_urls:
-            yield self.get_start_request(url)
+        if self.args.search_queries:
+            for url in self.start_urls:
+                meta: Dict[str, Any] = {
+                    "crawling_logs": {"page_type": "searchRequestTemplate"},
+                }
+                if self.args.crawl_strategy == EcommerceCrawlStrategy.full:
+                    meta["page_params"] = {"full_domain": get_domain(url)}
```

Comment:
I wonder what's the use case for this.

Is the idea that navigation model fails (misses some items, or misses pagination), and by picking "full" strategy we can follow almost any link on the target page, as a workaround? Do you know if it works in practice - won't you get products which are unrelated to the search query submitted - and if so, what's the point of using it?

Reply:
I cannot think of a use case. I’ll make the strategy incompatible with search queries.

Reply:
8389d7b